dask/array/slicing.py

```diff
@@ -15,6 +15,20 @@
 colon = slice(None, None, None)
 
 
+def _sanitize_index_element(ind):
+    try:
+        ind2 = int(ind)
```

Comment:
This is much too lenient, for example it will allow string indices and convert them to integers...